src/test/scala/com/cognite/sdk/scala/v1/fdm/instances/CogniteAssetsTest.scala

```diff
@@ -0,0 +1,82 @@
+package com.cognite.sdk.scala.v1.fdm.instances
+
+import cats.effect.unsafe.implicits.global
+import com.cognite.sdk.scala.common.RetryWhile
+import com.cognite.sdk.scala.v1.fdm.Utils
+import com.cognite.sdk.scala.v1.fdm.instances.InstanceDeletionRequest.NodeDeletionRequest
+import com.cognite.sdk.scala.v1.fdm.instances.NodeOrEdgeCreate.NodeWrite
+import com.cognite.sdk.scala.v1.fdm.views.ViewReference
+import com.cognite.sdk.scala.v1.{CommonDataModelTestHelper, File, FileDownloadInstanceId, FileDownloadLink, FileUploadInstanceId, InstanceId}
+import sttp.client3.UriContext
+
+import java.io.{BufferedInputStream, FileInputStream}
+
+@SuppressWarnings(
+  Array(
+    "org.wartremover.warts.NonUnitStatements",
+    "org.wartremover.warts.ThreadSleep",
+    "org.wartremover.warts.Equals"
+  )
+)
+class CogniteAssetsTest extends CommonDataModelTestHelper with RetryWhile {
+
+  it should "make it possible to retrieve file and associated upload link and download link using instance id" in {
+
+    val instanceId: InstanceId = InstanceId(space = Utils.SpaceExternalId, externalId = "file_instance_ext_id2")
+
+    val testFile = InstanceCreate(
+      items = Seq(
+        NodeWrite(
+          space = instanceId.space,
+          externalId = instanceId.externalId,
+          Some(Seq(EdgeOrNodeData(ViewReference("cdf_cdm", "CogniteFile", "v1"), None))
+          ),
+          None
+        )),
+      None,
+      None,
+      None,
+      None
+    )
+    val createdItem = testClient.instances.createItems(testFile).unsafeRunSync()
+    val retrievedItem =
+      retry[Seq[File]](testClient.files.retrieveByInstanceIds(Seq(instanceId)).unsafeRunSync())
+
+    val retrievedSingleItem =
+      retry[File](testClient.files.retrieveByInstanceId(instanceId).unsafeRunSync())
+
+    val uploadLinkFile =
+      retry[File](testClient.files.uploadLink(FileUploadInstanceId(instanceId)).unsafeRunSync())
+
+    val file = new java.io.File("./src/test/scala/com/cognite/sdk/scala/v1/uploadTest.txt")
```

Comment:
nit: file would fit better `src/test/resources`, would be also cleaner to read via resources api than from relative path. Since it's in source code repo it should be a small file and should be ok to include into jar

Reply:
I'll do that, good suggestion